fe/fe-core/src/main/java/com/starrocks/sql/DeletePlanner.java

```diff
@@ -52,7 +52,7 @@ public ExecPlan plan(DeleteStmt deleteStatement, ConnectContext session) {
         QueryRelation query = deleteStatement.getQueryStatement().getQueryRelation();
         List<String> colNames = query.getColumnOutputNames();
         ColumnRefFactory columnRefFactory = new ColumnRefFactory();
-        LogicalPlan logicalPlan = new RelationTransformer(columnRefFactory, session).transformWithSelectLimit(query);
+        LogicalPlan logicalPlan = new RelationTransformer(columnRefFactory, session).transform(query);
```

Comment:
why change this? Didn't get it from the description or this pr.

Reply:
delete/update/insert/create as select statement can't use sql_select_limit to limit result